pkg/frontend/adminactions/upgrade.go

```diff
@@ -50,7 +51,7 @@ func upgrade(ctx context.Context, log *logrus.Entry, configClient configclient.I
 		// Get Cluster upgrade version based on desired version
 		stream := version.GetUpgradeStream(streams, desired, upgradeY)
 		if stream == nil {
-			return fmt.Errorf("not upgrading: stream not found")
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Not upgrading: stream not found.")
```

Comment:
I'm OK with the general idea of using NewCloudError/http.StatusInternalServerError/api.CloudErrorCodeInternalServerError in this PR, but this specific case doesn't make sense to me.

I feel like it makes more sense to return nil here - the RP considered there was nothing to do, all good?

Reply:
I think it is more of "You have an error in your Stream config vs what is in the cluster CVO somewhere error here" 
I would better want to know when this happens and SRE raises it in slack than silent error.

Reply:
With this code the SRE will get a visible error if they try to upgrade a cluster which doesn't need to be upgraded, and I don't think that's a good idea.

Reply:
persuaded